### ISSUE TITLE
Widget is not rebuilt correctly for voice control after states change with iOS 16

### ISSUE THREAD

**Issue by jacobsimionato** [flutter/flutter#121770]
Copied from internal google issue b/271201186 - see that bug for a video of the repro case.

### Summary 

In the reproduction product, if the user says `2`, the widget should be rebuilt to show semantic label for each items in the PageView.

Note that: this is only happening on iOS 16, we can't reproduce this issue on iOS 15.

<details>
  <summary>Repro code</summary>
  
```
class TapboxA extends StatefulWidget {
  TapboxA({Key? key}) : super(key: key);
 
  @override
  _TapboxAState createState() => _TapboxAState();
}
 
class _TapboxAState extends State<TapboxA> {
  bool _active = false;
 
  final pageViewKey = GlobalKey();
  final pageController = PageController(initialPage: 0, viewportFraction: 0.15);
 
  void _handleTap() {
    setState(() {
      _active = !_active;
    });
  }
 
  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        const Spacer(),
        SizedBox(
          height: 88,
          child: _active
              ? _buildPageView(
                  context,
                )
              : Semantics(
                  label: 'PageView widget',
                  hint: 'hint',
                  button: true,
                  onTap: _handleTap,
                  excludeSemantics: true,
                  child: _buildPageView(
                    context,
                  ),
                ),
        )
      ],
    );
  }
 
  Widget _buildPageView(BuildContext context) {
    return PageView.builder(
      key: pageViewKey,
      controller: pageController,
      allowImplicitScrolling: true,
      itemCount: 20,
      itemBuilder: (_, index) => _buildItem(
        context,
        index: index,
      ),
    );
  }
 
  Widget _buildItem(
    BuildContext context, {
    required int index,
  }) {
    return Semantics(
      label: 'label',
      button: true,
      excludeSemantics: true,
      onTap: () async {
        await _animateToPage(index);
        _handleTap();
      },
      child: const Center(child: Text('text')),
    );
  }
 
  Future<void> _animateToPage(int page) {
    return pageController.animateToPage(
      page,
      duration: const Duration(milliseconds: 400),
      curve: Curves.easeInOut,
    );
  }
}class TapboxA extends StatefulWidget {
  TapboxA({Key? key}) : super(key: key);
 
  @override
  _TapboxAState createState() => _TapboxAState();
}
 
class _TapboxAState extends State<TapboxA> {
  bool _active = false;
 
  final pageViewKey = GlobalKey();
  final pageController = PageController(initialPage: 0, viewportFraction: 0.15);
 
  void _handleTap() {
    setState(() {
      _active = !_active;
    });
  }
 
  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        const Spacer(),
        SizedBox(
          height: 88,
          child: _active
              ? _buildPageView(
                  context,
                )
              : Semantics(
                  label: 'PageView widget',
                  hint: 'hint',
                  button: true,
                  onTap: _handleTap,
                  excludeSemantics: true,
                  child: _buildPageView(
                    context,
                  ),
                ),
        )
      ],
    );
  }
 
  Widget _buildPageView(BuildContext context) {
    return PageView.builder(
      key: pageViewKey,
      controller: pageController,
      allowImplicitScrolling: true,
      itemCount: 20,
      itemBuilder: (_, index) => _buildItem(
        context,
        index: index,
      ),
    );
  }
 
  Widget _buildItem(
    BuildContext context, {
    required int index,
  }) {
    return Semantics(
      label: 'label',
      button: true,
      excludeSemantics: true,
      onTap: () async {
        await _animateToPage(index);
        _handleTap();
      },
      child: const Center(child: Text('text')),
    );
  }
 
  Future<void> _animateToPage(int page) {
    return pageController.animateToPage(
      page,
      duration: const Duration(milliseconds: 400),
      curve: Curves.easeInOut,
    );
  }
}
```
</details>

**Comment by jmagman** [comment]
cc @hangyujin

**Comment by hangyujin** [comment]
This only happens when
1. Voice control on , but voice over is off
2. The widget is scrollable(e.g pageView, listView)
2. The scrollable widget is using the same manually assigned key before and after rebuilding although its parent is changed.

For the internal bug, a workaround is to assign different key to the active/inactive pageView.

**Comment by hangyujin** [comment]
mark this to P3 as the internal bug is commented as "unblocked for the next coming release"

**Comment by cyanglaz** [comment]
@hangyujin Do you still plan to work on this?

**Comment by hangyujin** [comment]
@cyanglaz  I'm not working on this
